editor/editor_node.h

```diff
@@ -218,13 +221,58 @@ class EditorNode : public Node {
 		TOOL_MENU_BASE = 1000
 	};
 
+	enum {
+		MAX_INIT_CALLBACKS = 128,
+		MAX_BUILD_CALLBACKS = 128
+	};
+
 	enum ScriptNameCasing {
 		SCENE_NAME_CASING_AUTO,
 		SCENE_NAME_CASING_PASCAL_CASE,
 		SCENE_NAME_CASING_SNAKE_CASE
 	};
 
-	SubViewport *scene_root; // root of the scene being edited
+	struct BottomPanelItem {
+		String name;
+		Control *control = nullptr;
+		Button *button = nullptr;
+	};
+
+	struct ExportDefer {
+		String preset;
+		String path;
+		bool debug = false;
+		bool pack_only = false;
+	} export_defer;
+
+	static EditorNode *singleton;
+
+	EditorCommandPalette *command_palette;
+	EditorData editor_data;
+	EditorExport *editor_export;
+	EditorFolding editor_folding;
+	EditorInterface *editor_interface;
+	EditorLog *log;
+	EditorNativeShaderSourceVisualizer *native_shader_source_visualizer;
+	EditorPlugin *editor_plugin_screen;
+	EditorPluginList *editor_plugins_force_input_forwarding;
+	EditorPluginList *editor_plugins_force_over;
+	EditorPluginList *editor_plugins_over;
+	EditorQuickOpen *quick_open;
+	EditorQuickOpen *quick_run;
+	EditorResourcePreview *resource_preview;
+	EditorRun editor_run;
+	EditorRunNative *run_native;
+	EditorSelection *editor_selection;
+	EditorSelectionHistory editor_history;
+	EditorSettingsDialog *editor_settings_dialog;
+
+	ProjectExportDialog *project_export;
+	ProjectSettingsEditor *project_settings_editor;
```

Comment:
These could be initialized to `nullptr` here.

Reply:
You're right, I changed it accordingly. But aren't there much more pointer member variables which are not initialized to nullptr (across the whole codebase)? So is there a general rule where to initialize to `nullptr` or should it ideally always be done?

Reply:
Ideally, it should always be done. If it's not done somewhere, it's usually old code or someone missed it during review. We have occasional crashes caused by uninitialized pointers or variables.

Reply:
We did a pass at initialization most member variables a while ago, but at the time we did not decide to initialize pointers to `nullptr`. So most of the codebase doesn't initialize them, but we can confirm with the other contributors that this should be done and then implement it with a regex.

Reply:
If you do `if (pointer)`, the behavior is undefined when the pointer is not initialized and results in crashes that occur randomly (I've seen two of them recently). I don't know if there is anything against initializing them, but I'd say it's better to be safe.